File: Ruby/problem_2.rb
answer = 2
fib = [1, 2]

loop do
  next_fib = fib[-2] + fib[-1]
  if next_fib > 4E6
    puts answer
    break
  end
  answer += next_fib if next_fib % 2 == 0
  fib.push(next_fib)
end

File: JS/problem2.js
var fib = function(n) {
  return (n < 2) ? n : fib(n-1) + fib(n-2);
};

var i   = 1,
    sum = 0;

while (fib(i) <= 4e6) {
  if (fib(i) % 2 === 0) {
    sum += fib(i);
  }
  i++;
}
